agent/structs/errors.go

```diff
@@ -47,3 +49,7 @@ func IsErrRPCRateExceeded(err error) bool {
 func IsErrServiceNotFound(err error) bool {
 	return err != nil && strings.Contains(err.Error(), errServiceNotFound)
 }
+
+func IsErrLeaderNotTracked(err error) bool {
+	return err != nil && strings.Contains(err.Error(), errLeaderNotTracked)
+}
```

Comment:
We should probably avoid adding more of these functions that use string comparison for detecting error types. Since this is only used in tests, it should be fine to remove it and use `errors.Is()`.

Reply:
That was my original inclination but since the other funcs existed I just added one for consistencies sake. I have removed it and replaced its usage with errors.Is